### ISSUE TITLE
2.1.1 prevents cache invalidation (flush)

### ISSUE THREAD

**Issue by NGMarmaduke** [storyblok/storyblok-ruby#17]
Since upgrading to 2.1.1 we are unable to flush our cached stories.

I've tracked down the issue to this commit: https://github.com/storyblok/storyblok-ruby/pull/14/commits/98e8e7869610fce17b29f6265e0fec5f21db736a

In particular this line: https://github.com/storyblok/storyblok-ruby/commit/98e8e7869610fce17b29f6265e0fec5f21db736a#diff-2bfa0e3e5211b947c48a9fe1dea26366R243

The `cache_version` instance variable is being set to '0' on initialization, it looks like the intention is that the cache version will automatically be updated with calls to `run_request` by fetching the cache version from the 'cv' attribute on API responses however it appears that 'cv' doesn't exist in the api responses we're receiving. The end result is that the cache version being stored in Redis keeps getting reset to '0' every time a story is requested - we can set the correct cache version by calling the flush method, but this is almost instantly being set back to '0' when a page is requested.

Not sure if there's additional configuration required to make caching work with the V2 API?

**Comment by 23tux** [comment]
We have the exact same problem. Is there any fix to this?

**Comment by jeantristan** [comment]
I've got exactly the same issue. Did you manage to fix this ?

**Comment by 23tux** [comment]
@jeantristan unfortunately, no :( We just downgraded the version

**Comment by SxDx** [comment]
Switching the client to `API Version 2` fixed it for me, as the `cv` key is present in the response.

```ruby
Storyblok::Client.new(
  api_version: 2,
  version: "published",
  token: token,
  cache: cache,
)  
```

**Comment by trappist** [comment]
Flush works, but gets repopulated by the same cached version from the cdn on a subsequent request

**Comment by christianzoppi** [comment]
Thanks @NGMarmaduke. This issue is fixed and closed by #21, with the update to the V2 API as the default version for the client, which has an enhanced caching strategy.

**Comment by pfeiffer** [comment]
Hey @christianzoppi!

We're seeing this issue with the new gem version and using the `v1` of the API. As it stands today, developers are able to set the API version when initializing the client, but setting it to `1` will result in `#flush` not working.

I think it that the caching and flushing should either be compatible with both v1 and v2 of the API _or_ support should be dropped for specifying the API version if the behavior of the gem cannot be kept the same between the two versions.
